packages/flame/lib/src/components/component.dart

```diff
@@ -590,10 +591,56 @@ class Component {
     return (parent is T ? parent : parent?.findParent<T>()) as T?;
   }
 
-  /// Called to check whether the point is to be counted as within the component
-  /// It needs to be overridden to have any effect, like it is in
-  /// PositionComponent.
-  bool containsPoint(Vector2 point) => false;
+  /// Checks whether the [point] is within this component's bounds.
+  ///
+  /// This method should be implemented for any component that has a visual
+  /// representation and non-zero size. The [point] is in the local coordinate
+  /// space.
+  bool containsLocalPoint(Vector2 point) => false;
+
+  /// Same as [containsLocalPoint], but for a "global" [point].
+  ///
+  /// This will be deprecated in the future, due to the notion of "global" point
+  /// not being well-defined.
+  bool containsPoint(Vector2 point) => containsLocalPoint(point);
+
+  /// An iterable of descendant components intersecting the given point. The
+  /// [point] is in the local coordinate space.
+  ///
+  /// More precisely, imagine a ray originating at a certain point (x, y) on
+  /// the screen, and extending perpendicularly to the screen's surface into
+  /// your game's world. The purpose of this method is to find all components
+  /// that intersect with this ray, in the order from those that are closest to
+  /// the user to those that are farthest.
+  ///
+  /// The return value is an [Iterable] of `(component, point)` pairs, which
+  /// gives not only the components themselves, but also the points of
+  /// intersection, in local coordinates.
+  ///
+  /// The default implementation relies on the [CoordinateTransform] interface
+  /// in order to translate from the parent's coordinate system into the local
+  /// one. Make sure that your component implements this interface if it alters
+  /// the coordinate system when rendering.
+  ///
+  /// If your component overrides [renderTree], then it almost certainly needs
+  /// to override this method as well, so that this method can find all rendered
+  /// components wherever they are.
+  Iterable<ComponentPoint> componentsAtPoint(Vector2 point) sync* {
```

Comment:
I'll write it here too so that it isn't forgotten; not completely happy with this name since it kind of states that it will return `Component`s, but it returns `ComponentPoint`s (I haven't come up with any alternative names yet though, `componentPointsAtPoint` is even worse). Not completely happy with having the `ComponentPoint` wrapper at all, but maybe there is no other way.